File: src/main/logs.js
import fs from 'fs'
import path from 'path'
import { app } from 'electron'

const logsPath = path.join(app.getPath('userData'), 'logs.json')

export function getLogs() {
  try {
    if (!fs.existsSync(logsPath)) {
      return []
    }
    const data = fs.readFileSync(logsPath, 'utf-8')
    return JSON.parse(data)
  } catch (err) {
    console.error('Error reading logs:', err)
    return []
  }
}

export function writeLog(level, context, message) {
  try {
    const logs = getLogs()
    const newEntry = {
      timestamp: new Date().toISOString(),
      level, // 'info' | 'error' | 'warning'
      context, // 'backup' | 'restore' | 'system'
      message
    }
    logs.push(newEntry)
    // Keep last 300 logs
    if (logs.length > 300) {
      logs.shift()
    }
    fs.writeFileSync(logsPath, JSON.stringify(logs, null, 2), 'utf-8')
  } catch (err) {
    console.error('Error writing log:', err)
  }
}

File: src-tauri/src/logs.rs
use serde::{Deserialize, Serialize};
use std::fs;
use std::path::Path;

#[derive(Serialize, Deserialize, Clone, Debug)]
pub struct LogEntry {
    pub timestamp: String,
    pub level: String,   // "info" | "error" | "warning"
    pub context: String, // "backup" | "restore" | "system"
    pub message: String,
}

pub fn get_logs(logs_path: &Path) -> Vec<LogEntry> {
    if !logs_path.exists() {
        return Vec::new();
    }
    match fs::read_to_string(logs_path) {
        Ok(content) => serde_json::from_str(&content).unwrap_or_else(|_| Vec::new()),
        Err(_) => Vec::new(),
    }
}

pub fn write_log(logs_path: &Path, level: &str, context: &str, message: &str) {
    let mut logs = get_logs(logs_path);
    
    let now = chrono::Local::now().to_rfc3339();
    let new_entry = LogEntry {
        timestamp: now,
        level: level.to_string(),
        context: context.to_string(),
        message: message.to_string(),
    };
    
    logs.push(new_entry);
    if logs.len() > 300 {
        logs.remove(0);
    }
    
    if let Ok(serialized) = serde_json::to_string_pretty(&logs) {
        let _ = fs::write(logs_path, serialized);
    }
}
